### PR TITLE
[Feature] Customizable request schema

### PR DESCRIPTION
**Pull request by bojiang**
## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [ ] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [x] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [ ] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [x] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [x] My change reduces project test coverage and requires unit tests to be added
- [x] I have added unit tests covering my code change
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: protos/repository.proto [hunk 1]
```diff
@@ -50,6 +50,7 @@ message BentoServiceMetadata {
     int32 mb_max_latency = 7;
     int32 mb_max_batch_size = 8;
     bool batch = 9;
+    string route = 10;
```
Comment (yubozhao) [inline protos/repository.proto:53]: I think this is part of the #1334 ?  I guess we are holding this after that PR merge in?

**Comment by Trollgeir** [comment]
Might I suggest supporting [pydantic](https://pydantic-docs.helpmanual.io/) for model schemas? It'll give BentoML field validation/errors/schema out of the box.

**Comment by bojiang** [comment]
> Might I suggest supporting pydantic for model schemas? It'll give BentoML field validation/errors/schema out of the box.
@Trollgeir Appreciate it. This PR is just minimal changes to make it possible.

**Comment by bojiang** [comment]
@yubozhao ready for review

**Comment by yubozhao** [comment]
LGTM. Merging it

@Trollgeir I think that's a great suggestion.  Let's create a different issue and PR to track include Pydantic for validation

**Pull request merged**